plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -71,6 +71,7 @@
 
     private DataSize maxSplitSize = DataSize.of(64, MEGABYTE);
     private int maxPartitionsPerScan = 100_000;
+    private int maxPartitionsForEagerLoad = 100_000;
```

Comment:
Add an `@AssertTrue` validator method that verifies `maxPartitionsForEagerLoad  <= maxPartitionsPerScan `